contrib/clojure-package/README.md

```diff
@@ -0,0 +1,203 @@
+Do `lein run` for the cpu version or `lein run :gpu` for gpu.
+
+## Generating documentation
+
+To generate api docs, run `lein codox`. The html docs will be generated in the target/docs directory.
+
+_Note: There is an error thrown in the generated code due to some loading issues, but the docs are all still there._
+
+## Code Coverage
+
+To run the Code Coverage tool. Run `lein cloverage`.
+
+## FAQ
+
+
+**Why build on the Scala package?**
+
+The motivation section addresses this, but the main reason is high leverage is using the great work that the Scala package has already done.
+
+**How can I tell if the gpu is being used?**
+I find this command to be very handy
+
+`nvidia-smi --query-gpu=timestamp,name,utilization.gpu,utilization.memory,memory.total,memory.free,memory.used --format=csv -l 5
+timestamp, name, utilization.gpu [%], utilization.memory [%], memory.total [MiB], memory.free [MiB], memory.used [MiB]`
+
+**Is the Gluon Api supported?**
+There are 3 high level apis supported in MxNet: FeedForward, Module, and Gluon. The Module api is supported in the Clojure package because of the existing support for it in the Scala package. The Module api is very similar to the Gluon api and examples of the usage can be found in the examples directory.
+
+Gluon support will come later and may or may not be built on the Scala gluon api (when it lands there)
+
+## Architecture & Design
+
+See the Confluence page: https://cwiki.apache.org/confluence/display/MXNET/MXNet+Clojure
+
+## Building and Deploying Jars
+The process to build and deploy the jars currently is a manual process using the `lein` build tool and `Clojars`, the Clojure dependency hosting platform.
+
+There is one jar for every system supported.
+
+- Comment out the line in the `project.clj` for the system that you are targeting, (example OSX cpu you would uncomment out ` [org.apache.mxnet/mxnet-full_2.11-osx-x86_64-cpu "1.2.0"]` but leave the linux deps commented)
+- Change the `defproject org.apache.mxnet.contrib.clojure/clojure-mxnet "0.1.1-SNAPSHOT"` in the project to reference the correct version number and jar description. For example changing the line to be `org.apache.mxnet.contrib.clojure/mxnet-osx-cpu "0.1.2"` would create a jar with the group id of `org.apache.mxnet.contrib.clojure` and the artifact name of `mxnet-osx-cpu` and the version of `0.1.2`
+- Run `lein clean`
+- Run `lein jar` to create the jar
+- Check that the jar looks alright in the `/target` directory.
+
+To deploy the jar to Clojars, you do `lein deploy clojars` and it will prompt you for your username and password.
+
+_Note: Integration with deployment to Nexus can be enabled too for the future [https://help.sonatype.com/repomanager2/maven-and-other-build-tools/leiningen](https://help.sonatype.com/repomanager2/maven-and-other-build-tools/leiningen)_
+
+You would repeat this process out on the AWS Deep Learning AMI, once for the linux cpu and once for the linux gpu.
+
+
+### Deferred
+* Feed Forward API
```

Comment:
I had asked about it earlier in the development cycle and didn't seen any of the online docs for the Scala API pointing to it. It was said that it was deprecated but still supported https://mxnet.incubator.apache.org/api/scala/index.html so I went the route of supporting Module first instead of the Feed Foward (Model) API.